Library/Homebrew/test/migrator_spec.rb

```diff
@@ -0,0 +1,269 @@
+require "migrator"
+require "test/support/fixtures/testball"
+require "tab"
+require "keg"
+
+class Migrator
+  describe self do
+    let(:new_formula) { Testball.new("newname") }
+    let(:old_formula) { Testball.new("oldname") }
+
+    before(:each) do
+      allow(new_formula).to receive(:oldname).and_return("oldname")
+    end
+
+    it "raises an error if there is no old name" do
+      expect {
+        described_class.new(old_formula)
+      }.to raise_error(MigratorNoOldnameError)
+    end
+
+    it "raises an error if there is no old path" do
+      expect {
+        described_class.new(new_formula)
+      }.to raise_error(MigratorNoOldpathError)
+    end
+
+    it "raises an error if the Taps differ" do
+      keg = HOMEBREW_CELLAR/"oldname/0.1"
+      keg.mkpath
+      tab = Tab.empty
+      tab.tabfile = HOMEBREW_CELLAR/"oldname/0.1/INSTALL_RECEIPT.json"
+      tab.source["tap"] = "homebrew/core"
+      tab.write
+
+      expect {
+        described_class.new(new_formula)
+      }.to raise_error(MigratorDifferentTapsError)
+    end
+  end
+
+  describe self do
```

Comment:
I can't think of a better way to make sure the test files aren't created for the error tests.